### ISSUE TITLE
Rails 4: after_commit callbacks do not work for ActiveType::Object

### ISSUE THREAD

**Issue by mattbrictson** [makandra/active_type#47]
When `ActiveType::Object` is used in Rails 4, `before_save` and `after_save` callbacks are executed as expected, but `after_commit` is not. This used to work in Rails 3.2.

* active_type (0.4.0)
* pg (0.18.2)

Trivial example:

```ruby
class Hello < ActiveType::Object
  after_commit :say_hi

  private

  def say_hi
    puts "Hi!"
  end
end
```

When this model is saved, the `say_hi` method is never executed.

```
$ rails c 
Loading development environment (Rails 4.0.13)
>> h = Hello.new
=> #<Hello >
>> h.save!
   (0.1ms)  BEGIN
   (0.1ms)  COMMIT
=> true
>> 
```

**Comment by mattbrictson** [comment]
@henning-koch Regarding the usefulness of `after_commit`:

Since `save` implicitly creates a transaction, you don't want to invoke any potentially long-running operations, like an external API call, during that transaction. So `after_commit` is useful in that scenario for performance reasons.

It is also especially useful for things like triggering email notifications, since you want to make absolutely sure that the commit succeeded (e.g. no unique constraint violations) before sending an email.

Finally, if you are enqueuing a background job (e.g. via Sidekiq) as part of the `save`, if you don't use `after_commit` then you will run into a race condition where the background job could execute before the database transaction is committed. This is such a common problem that it is mentioned in the [Sidekiq FAQ](https://github.com/mperham/sidekiq/wiki/FAQ#why-am-i-seeing-a-lot-of-cant-find-modelname-with-id12345-errors-with-sidekiq).

What makes `ActiveType::Object` attractive is that it can be used to represent a "service/command object" that orchestrates many persistent AR objects but where the orchestration itself is not a database record. It would be nice to be able to use `after_commit` in these situations.

**Comment by mattbrictson** [comment]
I've figured out that the following change is what is needed to fix this bug. I'll open a PR once I have a test written.

```diff
diff --git a/lib/active_type/no_table.rb b/lib/active_type/no_table.rb
index 2e840d9..7831193 100644
--- a/lib/active_type/no_table.rb
+++ b/lib/active_type/no_table.rb
@@ -77,6 +77,7 @@ module ActiveType
       end
     else
       def _create_record(*)
+        @new_record = false
         true
       end
```

**Comment by kratob** [comment]
Thanks for the fix!
I've released the change as version 0.4.4.
